docker-compose-search.yml

```diff
@@ -0,0 +1,35 @@
+# docker-compose-search.yml starts "search" service and override some
```

Comment:
Why don't we just put elasticsearch into the same `docker-compose.yml` so that the same compose will setup search as well? Perhaps I'm missing something.

Reply:
The only reason to have it in another yaml file is because elasticsearch consumes a lot of RAM and most of the time is not needed for development. So, I'm doing it optional to start.

I wanted to have a better pattern, something like `docker-compose up --search` or similar but I didn't find a how to do it.

Reply:
You can limit the ES memory consumption with `"ES_JAVA_OPTS=-Xms512m -Xmx512m"` environment variable, I don't think you can provide custom arguments or conditional container spin up to docker-compose

Reply:
Thanks for the tip @yakky! I've done that at https://github.com/readthedocs/readthedocs.org/pull/6295/commits/6acd6f45c299d8a2c53646bbcb7148b16fd00511

Reply:
Basically now you can run everything with search by calling:

```
docker-compose -f docker-compose.yml -f docker-compose-search.yml up
```

or without search

```
docker-compose up
```